python/lsst/ctrl/oods/gen3ButlerIngester.py

```diff
@@ -40,20 +56,77 @@ def __init__(self, butlerConfig):
 
         instr = getInstrument(instrument)
         run = instr.makeDefaultRawIngestRunName()
-        opts = dict(run=run, writeable=True)
-        self.btl = Butler(butlerConfig, **opts)
+        opts = dict(run=run, writeable=True, collections=self.collections)
+        self.butler = Butler(butlerConfig, **opts)
 
         if register:
             # Register an instrument.
-            instr.register(self.btl.registry)
+            instr.register(self.butler.registry)
 
     def ingest(self, filename):
+        """Ingest a file into a butler
+
+        Parameters
+        ----------
+        filename: `str`
+            file name of the file to ingest
+        """
 
         # Ingest image.
         cfg = RawIngestConfig()
         cfg.transfer = "direct"
-        task = RawIngestTask(config=cfg, butler=self.btl)
+        task = RawIngestTask(config=cfg, butler=self.butler)
         task.run([filename])
 
     def getName(self):
+        """Return the name of this ingester
+
+        Returns
+        -------
+        ret: `str`
+            name of this ingester
+        """
         return "gen3"
+
+    async def run_task(self):
+        """run the clean() method at the configured interval
+        """
+        seconds = TimeInterval.calculateTotalSeconds(self.scanInterval)
+        while True:
+            self.clean()
+            await asyncio.sleep(seconds)
+
+    def clean(self):
+        """Remove all the datasets in the butler that
+        were ingested before the configured Interval
+        """
+
+        # calculate the time value which is Time.now - the
+        # "olderThan" configuration
+        t = Time.now()
+        interval = collections.namedtuple("Interval", self.olderThan.keys())(*self.olderThan.values())
+        td = TimeDelta(interval.days*u.d + interval.hours * u.h +
+                       interval.minutes*u.min + interval.seconds*u.s)
+        t = t - td
+
+        # get the datasets
+        ref = self.butler.registry.queryDatasets(datasetType=...,
+                                                 collections=self.collections,
+                                                 where=f"ingest_date < T'{t}'")
```

Comment:
I think that in the most recent weekly you can use `bind` (it was broken before then) to stop having to get the quoting right, although I have no idea whether it would work in the context. Something like:

```python
 ref = self.butler.registry.queryDatasets(datasetType=...,
                                          collections=self.collections,
                                          where=f"ingest_date < ref_date", bind={"ref_date": t})
```

cc/ @TallJimbo in case I'm giving bad advice.

Reply:
That's supposed to work if `t` is a `astropy.time.Time`, but the ingest timestamp is a unique column as the only non-TAI time we've got, so even if it doesn't raise, you may want to make sure you're not off by 37s.

Reply:
Bind worked.  This was fixed.